packages/api/src/ai/services/ai.service.ts

```diff
@@ -16,13 +16,42 @@ export class AiService {
     });
   }
 
-  async askAiInFreeText(input: string, chatId: string) {
-    const chain = this.chainService.getChain(chatId, this.llmModel);
+  async askAiInFreeText(
+    input: string,
+    chatId: string,
+    shouldSummarize: boolean
+  ) {
+    let summary;
+
+    if (shouldSummarize) {
+      try {
+        summary = await this.askAiToSummarize(chatId);
+      } catch (e) {
+        console.log(e);
+      }
+    }
+    const chain = this.chainService.getChain(
+      chatId,
+      this.llmModel,
+      summary.response
+    );
 
     try {
       return await chain.call({ input });
     } catch (e) {
       console.error(e.response.data);
     }
   }
+
+  async askAiToSummarize(chatId: string) {
```

Comment:
```suggestion
  private async askAiToSummarize(chatId: string) {
```

I don't want this method to be exposed to the outside world of the AI service. The AI service is the Facade API to whoever wishes to use it. For now the API exposes the `askAiInFreeText` method, and no other :)